core/src/main/java/me/aurium/branch/annotate/marker/permission/Permission.java

```diff
@@ -0,0 +1,34 @@
+/*
+ *
+ *  Branch
+ *  Copyright © 2021 Aurium
+ *
+ *  Branch is free software: you can redistribute it and/or modify
+ *  It under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  Branch is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with Branch. If not, see <https://www.gnu.org/licenses/>
+ *  and navigate to version 3 of the GNU Affero General Public License.
+ *
+ */
+
+package me.aurium.branch.annotate.marker.permission;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+@Target({ElementType.METHOD,ElementType.CONSTRUCTOR})
+public @interface Permission {
```

Comment:
Annotations like this one are examples of why I greatly prefer code-driven libraries to annotations-as-code. Here, the attributes of the annotation are hard-coded in the use of it and type-safety is lost by presumably instantiating the relative checker.

Reply:
The entire framework is code-driven. The annotation system exists purely if you want to use it (it wraps the rest of the framework. The permission interface basically just instantiates a StringPermission.)